File: app/src/main/java/com/cyberegylet/antiDupeGallery/models/Album.java
package com.cyberegylet.antiDupeGallery.models;

import androidx.annotation.NonNull;

import com.cyberegylet.antiDupeGallery.backend.FileManager;

import java.io.File;
import java.util.Objects;

public class Album
{
	private ImageFile indexImage;
	private String name;
	private File file;
	private long modifiedDate;
	private long size;
	private long count;
	private boolean isHidden;
	private final long id;
	private static long countId = 0;

	public Album(File file)
	{
		setFile(file);
		this.id = countId;
		countId++;
	}

	public Album(Album folder, boolean copyImages)
	{
		this.name = folder.name;
		this.file = folder.file;
		this.modifiedDate = folder.modifiedDate;
		this.id = folder.id;
		if (copyImages)
		{
			indexImage = folder.indexImage;
			this.size = folder.size;
			this.count = folder.count;
		}
		else
		{
			this.size = 0;
			this.count = 0;
		}
	}

	public Album(String path) { this(new File(path)); }

	public void addImage(ImageFile imageFile)
	{
		if (count == 0) indexImage = imageFile;
		size += FileManager.getSize(imageFile.getFile());
		count++;
	}

	public ImageFile getIndexImage() { return indexImage; }

	@NonNull
	public String getName() { return name; }

	public File getFile() { return file; }

	public String getPath() { return file.getPath(); }

	public long getSize() { return size; }

	public long getCount() { return count; }

	public long getModifiedDate() { return modifiedDate; }

	public long getCreationDate() { return modifiedDate; }

	public boolean isHidden() { return isHidden; }

	public long getId() { return id; }

	public void setFile(File file)
	{
		String stringPath = Objects.requireNonNull(file.getPath());
		this.file = file;
		this.name = file.getName();
		this.isHidden = stringPath.contains("/.");
		this.modifiedDate = file.lastModified();
	}
}

File: app/src/main/java/com/cyberegylet/antiDupeGallery/models/Album.kt
package com.cyberegylet.antiDupeGallery.models

import com.cyberegylet.antiDupeGallery.backend.FileManager
import java.io.File

class Album : FileEntry
{
	var indexImage: ImageFile? = null
		private set
	var count: Long = 0
		private set

	constructor(file: File) : super(file, countId)
	{
		countId++
	}

	constructor(folder: Album, copyImages: Boolean)
	{
		name = folder.name
		file = folder.file
		modifiedDate = folder.modifiedDate
		id = folder.id
		if (copyImages)
		{
			indexImage = folder.indexImage
			size = folder.size
			count = folder.count
		}
		else
		{
			size = 0
			count = 0
		}
	}

	constructor(path: String) : this(File(path))

	fun addImage(imageFile: ImageFile)
	{
		if ((indexImage?.modifiedDate ?: 0) < imageFile.modifiedDate) indexImage = imageFile
		size += FileManager.getSize(imageFile.file)
		count++
		imageFile.isHidden = isHidden
	}

	override fun mySetFile()
	{
		super.mySetFile()
		modifiedDate = file.lastModified()
	}

	companion object
	{
		private var countId: Long = 0
	}
}